Projects/Contracts/Sources/Contracts.swift

```diff
@@ -83,3 +105,29 @@ extension Contracts: Tabable {
         )
     }
 }
+
+public struct ContractFetcher {
```

Comment:
Couldn't this just be an `effect`? We would then need to add some kind of key or id to each effect to cancel them in presentation, but that is something we should do I think.

Reply:
do you think this is worth doing already?